utils/python/CIME/XML/entry_id.py

```diff
@@ -77,10 +81,22 @@ def _get_value_match(self, node, attributes={}):
                     else:
                         match_count = 0
                         break
-            if match_count > 0 and match_count > match_max:
-                match_max = match_count
-                match_value = valnode.text
-
+            if match_count > 0:
+                # append the current result
+                if values.get("modifier") == "additive":
```

Comment:
These modifiers are currently only used on config_compsets.xml, the specific code could be moved to components.py if we wanted to make sure that continued to be the case.